src/gbt_model_builder.h

```diff
@@ -22,10 +22,12 @@
 #include <daal.h>
 #include "onedal/version.hpp"
 
-#if (((MAJOR_VERSION == 2023) && (MINOR_VERSION >= 2)) || (MAJOR_VERSION > 2023))
-#define _gbt_inference_has_missing_values_support 1
+#if (((MAJOR_VERSION == 2024) && (MINOR_VERSION >= 1)) || (MAJOR_VERSION > 2024))
+    #define _gbt_inference_api_version 2
+#elif (((MAJOR_VERSION == 2023) && (MINOR_VERSION >= 2)) || (MAJOR_VERSION > 2023))
+    #define _gbt_inference_api_version 1
 #else
-#define _gbt_inference_has_missing_values_support 0
+    #define _gbt_inference_api_version 0
 #endif
```

Comment:
I suggest adding some comments about the meaning of this numbers. Like "version 1 added support of missing values, version 2 added support of shap-values".